portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,18 +45,36 @@
     * @throws IOException if there is an I/O error
     */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
         Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
         List<Comment> comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
-            Comment comment = new Comment(entity.getKey().getId(), (String) entity.getProperty("comment"), (long) entity.getProperty("timestamp"));
+            Comment comment = new Comment(entity.getKey().getId(), (String) entity.getProperty("nickname"), (String) entity.getProperty("comment"), (long) entity.getProperty("timestamp"));
             comments.add(comment);
         }
-        Gson gson = new Gson();
-        response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(comments));
+        request.setAttribute("comments", comments);
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            String loginUrl = userService.createLoginURL("/data");
+            request.setAttribute("loginStatus", false);
+            request.setAttribute("loginUrl", loginUrl);
+        }
+        else {
```

Comment:
The Google Java style guide for if/else says to put the else on the same line as the preceding close brace.  I don't know if that's the style guide you're trying to follow here though. :)